tests/cram.t/run.t

```diff
@@ -43,44 +43,52 @@ combinations of produce-models et al.
 First, if model generation is not enabled, we should error out when a
 `(get-model)` statement is issued:
 
-  $ echo '(get-model)' | alt-ergo -i smtlib2 -o smtlib2 --continue-on-error
+  $ echo '(set-logic ALL)(check-sat)(get-model)' | alt-ergo -i smtlib2 -o smtlib2 --continue-on-error 2> /dev/null
+  
+  unknown
   (error "Model generation disabled (try --produce-models)")
 
 This should be the case Tableaux solver as well:
 
-  $ echo '(get-model)' | alt-ergo --sat-solver Tableaux -i smtlib2 -o smtlib2 --continue-on-error
+  $ echo '(set-logic ALL)(check-sat)(get-model)' | alt-ergo --sat-solver Tableaux -i smtlib2 -o smtlib2 --continue-on-error 2> /dev/null
+  
+  unknown
   (error "Model generation disabled (try --produce-models)")
 
 The messages above mention `--produce-models`, but we can also use
 `set-option`.
 
-  $ echo '(get-model)' | alt-ergo --produce-models -i smtlib2 -o smtlib2 --continue-on-error
```

Comment:
> No they are not skipped, they just result in an error; why do you think they should have been broken?

I thought that `(get-model)` would have returned an error about `set-logic` not being present rather than an error about the model not being here, but it looks like Dolmen does not complain about a missing `set-logic` for `get-model`.

> Or are you saying we want this test to show that alt-ergo will fail on bad smt files? This sounds relevant.

Yes, that was what the test was doing (although the error was different; it was "No model produced" and now it should be "Invalid command in start mode").

(Now thinking that we should probably also have a test for `(set-logic ALL)(assert false)(check-sat)(get-model)` for the "No model produced" error, I don't think we have that one)

Reply:
> I thought that `(get-model)` would have returned an error about `set-logic` not being present rather than an error about the model not being here, but it looks like Dolmen does not complain about a missing `set-logic` for `get-model`.

Not by default indeed. The situation is that, the dolmen functions to typecheck things would raise a missing set-logic error if called, but in the case of `get-model`, nothing needs to be typed. For that, dolmen has a `Flow` module, whose role is to check that the statements respect the smtlib constraints, see https://github.com/Gbury/dolmen/pull/135

Reply:
Ah! Interesting! I thought for a moment that we could maybe use that in this PR, but it looks like it is not designed to be extensible — it doesn't look like we can access the current mode from the state, or otherwise configure custom mode-specific filters (e.g. for custom options that can only be set in start mode), is that correct?

Reply:
OK. I think if Dolmen already had everything we needed it would have made sense to use that, but given that we would need to modify the `Flow` module to suit our needs and that we have an already working implementation inside Alt-Ergo in this PR, I am fine with either solution.

Reply:
> (Now thinking that we should probably also have a test for (set-logic ALL)(assert false)(check-sat)(get-model) for the "No model produced" error, I don't think we have that one)

This would return a different error: after check-sat we would be in `Unsat` mode, which is an invalid mode for `get-model`